website/docs/error-reference.md

```diff
@@ -1638,6 +2061,55 @@ def foo(x, y)
 end
 ```
 
+## 7035
+
+Sorbet can sometimes detect when a method is passed a block despite not
+accepting a block.
+
+```ruby
+# typed: strict
+extend T::Sig
+
+sig {void}
+def takes_no_block; end
+
+# BAD EXAMPLE
+takes_no_block {} # error: does not take a block
+```
+
+Why only sometimes? Technically all methods in Ruby are allowed to accept
+blocks. (Unlike positional arguments, if a method does not declare that it
+accepts a block argument but a caller passes one anyways the Ruby VM does not
```

Comment:
I kind of want a comma after "anyways"?  Not sure.

Reply:
thought about it for a while, with a comma sounds more weird in my head htan without. maybe we should rewrite the whole sentence, but i'm just going to punt.